File: dcp-solutions/2024-11-11/solution.c
```c
#include <stdlib.h>

int uniquePaths(int N, int M) {
    int dp[2][M + 1] = {{0}};
    
    for (int i = 0; i < N; i++) {
        int *temp = &dp[i % 2][0];
        temp[i / 2] = 1;
        
        for (int j = 1; j <= M; j++) {
            if (i == 0 && j == 0) continue;
            temp[j] += dp[(i + 1) % 2][j - 1] + dp[i % 2][j];
        }
    }
    
    return dp[N % 2][M];
}
```

File: dcp-solutions/2024-11-11/solution.js
```javascript
function uniquePaths(N, M) {
    let dp = Array(N).fill(0).map(() => Array(M).fill(1));
    
    for (let i = 1; i < N; i++) {
        for (let j = 1; j < M; j++) {
            dp[i][j] = dp[i - 1][j] + dp[i][j - 1];
        }
    }

    return dp[N - 1][M - 1];
}
```